### ISSUE TITLE
Announcement

### ISSUE THREAD

**Issue by Andrews54757** [AJS-development/Ogar-unlimited#1448]
I have noticed the amount we raised, and I think ogarUL is gonna go bye bye. Here is my plan.

### For OgarUL
* no more support
* still be there but no issues
* ogar websites shutdown. (probably a month after the ogarUL shutdown)

### For the new project
* No community support
* limited release

##### Features
* Fast
* uses clusters
* Graphical interface (to control server)
* client-server protection using key
* and more

Note that the new project would only be given to select people

**Comment by Agarian** [comment]
FUCK!

**Comment by Agarian** [comment]
@AJS-development/ogar-unlimited-devs please visit http://github.com/Agarian/Ogar-unlimited

**Comment by ZfsrGhS953** [comment]
Well, good luck with your project! Meanwhile, I have a server over there (sorry, no Git support) http://www.agarserv.com/forums/topic/some-petridish-pw-modes-in-ogar/ I already ported the most fun feature of Ogar Unlimited: MINIONS! Other most adequate features are going to be put in there too.

**Comment by SharkFinPro** [comment]
@Andrews54757 Could i help develop new version?

**Comment by raphyjb** [comment]
idk why each time i check in issues i still cant believe this one is still up.. all a joke right?
HOPE ITS NOT FUCKIN GOING DOWN OR ILL KYS :( THATS ALL I DO LEGIT ALL DAY.. PLEASE KEEP THE SITE UP ALTEAST @LegitSoulja

**Comment by firelightning13** [comment]
> Note that the new project would only be given to select people

Is it only for developers/contributors? @Andrews54757

**Comment by Agarian** [comment]
i have the site saved as a .html

**Comment by SharkFinPro** [comment]
@raphyjb I might be able to keep the site running....

**Comment by SharkFinPro** [comment]
@Andrews54757 Could i host the ogarUL site after it goes down?

**Comment by LegitSoulja** [comment]
@Andrews54757 
Nice to know. After charity water donations are over the site will automatically shut down, and also statistics, and then soon ogarul.tk will be removed and we will get a new domain. (After a month has passed, and before release)

@SharkFinPro yes, anyone can host a site. Just download the client files Andrew provided a long time ago.

@raphyjb If there was to ever be a new site, and or client it will be removed to conserve space, and site traffic, or will be redirected to the new client. I won't maintain it, and plus ogarul.tk is a free domain and will only last for a year before it's gone. So I won't no longer be responsible, and I won't register it again .

@firelightning What that means is only a selected few people will be able to obtain the future releases. I don't think all AJS members, or contributors will get access.

When the site shut down, everyone will see a sorry message on site when day 0 hits. All AJS members are listed as assisted in OgarUL, and also a link back to this github repo.

As most know, I deleted Skype so there's only 1 way to contact me and that's by my email on my github profile. I will no longer be distracted if I'm working, and will only check messages on my spare time. I check mail daily so you'll get a reply if you do contact me :). 

Anyways, I hope everyone lives a good life, stay focused in school, and like always there's many things in life you can do other than being worried about this. Majority of the people aren't even using ogarul but rather multiogar etc. I don't know why there's a lot of complaining :/

Just be happy the world is not ending yet. That what we all should worry about #Life. Yolo, so don't stress.

Sincerely,
- LegitSoulja

**Comment by SharkFinPro** [comment]
@LegitSoulja Could i have the plzy.ogarult.tk files? the files andrew provided don't have skins :/

**Comment by LegitSoulja** [comment]
You can't have the files, but you can get the skins.

**Comment by LegitSoulja** [comment]
And you can also re-register ogarul.tk if you want. After a few days after the site is down I'll terminate it.

**Comment by SharkFinPro** [comment]
@LegitSoulja could i have this file to learn how it works?
http://stats.ogarul.tk/getip.php

**Comment by LegitSoulja** [comment]
@SharkFinPro I recommend you to learn php because it's very simple to get a remote address of the connected client(browser).

**Comment by SharkFinPro** [comment]
where are the most recent client files? i tried ajs-development/client but it has no skins

**Comment by LegitSoulja** [comment]
@SharkFinPro that github page doesn't have skins, but it pulls the skins from play.ogarul.tk instead. Which I think no longer work due to the skins folder change.

**Comment by SharkFinPro** [comment]
@LegitSoulja I Know PHP i just want to see source code of that page :/
Is there a client that does have skins?

**Comment by LegitSoulja** [comment]
@SharkFinPro well, you should already know how to obtain the client with no skins. You already know about 2 locations where the client files are public.

**Comment by SharkFinPro** [comment]
I know about 2:
ajs-development/client
some issue has a bunch - but i forget which one

**Comment by SharkFinPro** [comment]
can i obtain a client _with_ skins?

**Comment by LegitSoulja** [comment]
@SharkFinPro sorry, I misread your comment. No, there's no public client with skins, but I have said I could provide you with the skins, but not the client files of play.ogarul.tk because it will not work on another header.

**Comment by SharkFinPro** [comment]
ok

**Comment by SharkFinPro** [comment]
could you give me one with the skins?

**Comment by SharkFinPro** [comment]
i could make play.ogarul.tk work on other headers :/

**Comment by LegitSoulja** [comment]
@SharkFinPro what's your email? Is it the same email on forum.ogarul.tk when you registered?

And what I mean about the files will not work on other headers which means if the client files where to ever be stolen, it will be redirected to play.ogarul.tk due to the security fixes I've applied for the kid leechers.

**Comment by SharkFinPro** [comment]
sharkfinpro@hotmail.com

**Comment by LegitSoulja** [comment]
@SharkFinPro I'll send you the skins there shortly.

**Comment by SharkFinPro** [comment]
also, why does ogarul.tk redirect to play.ogarul.tk? it used to not :(

**Comment by LegitSoulja** [comment]
I was experimenting with it, and it was pointless to have. So I had it redirect there for a faster way to get to play.ogarul.tk without having to type much.

**Comment by SharkFinPro** [comment]
oh, ok

**Comment by LegitSoulja** [comment]
It used to redirect to the forum before it turned into something expiremental. Now it doesn't.

**Comment by SharkFinPro** [comment]
ok

**Comment by SharkFinPro** [comment]
You made it go to the forum now :)

**Comment by LegitSoulja** [comment]
Yes, correct.

**Comment by SharkFinPro** [comment]
ok.

**Comment by SharkFinPro** [comment]
Also, where do your register your domains? i use freenom

**Comment by LegitSoulja** [comment]
@SharkFinPro i'll provide a public link to the skins in a sec, instead of sending it to your email.

**Comment by LegitSoulja** [comment]
@SharkFinPro http://dev.ogarul.tk/data/public/af6bd1

**Comment by LegitSoulja** [comment]
and the source to getip.php is
```
header('Content-Type: application/json');
$ip= isset($_SERVER['HTTP_X_FORWARDED_FOR']) ? $_SERVER['HTTP_X_FORWARDED_FOR'] : $_SERVER['REMOTE_ADDR'];
echo '{"ip":"'.$ip.'"}';
die();
```

**Comment by SharkFinPro** [comment]
k

**Comment by SharkFinPro** [comment]
How do i mass download?

**Comment by LegitSoulja** [comment]
@SharkFinPro there should be a download button to download that entire folder.

**Comment by SharkFinPro** [comment]
Ok...it seems i can only download 1 at a time though

**Comment by SharkFinPro** [comment]
![capture](https://cloud.githubusercontent.com/assets/18649707/18227897/58790f1c-71ea-11e6-9729-dc37505a36aa.JPG)

**Comment by LegitSoulja** [comment]
@SharkFinPro maybe there's a file limit because there's more than 200 skins. 

http://play.ogarul.tk/skins.zip

**Comment by SharkFinPro** [comment]
That link goes to nowhere :/

**Comment by LegitSoulja** [comment]
Try again lol

**Comment by SharkFinPro** [comment]
k now it does :P

**Comment by SharkFinPro** [comment]
i have the client from github repo, do i just put all skins in the skins folder?

**Comment by LegitSoulja** [comment]
yes, and make sure in main_out.js the skin folder is correct.

**Comment by SharkFinPro** [comment]
?

**Comment by LegitSoulja** [comment]
@SharkFinPro The folder you put the skins into, make sure in main_out.js the variable SKIN_URL is the folder the skins are contained within, or they will not load if that path isn't correct..

**Comment by SharkFinPro** [comment]
Ok

**Comment by LegitSoulja** [comment]
its currently
```
var SKIN_URL = "http://play.ogarul.tk/skins/";
```

needs to be changed to
```
var SKIN_URL = "/skins/";

**Comment by SharkFinPro** [comment]
ok

**Comment by SharkFinPro** [comment]
where can i get the nicer main page for play.ogarul.tk? is there differrent code than ajs-development/client, or do i have ot recreate it?

**Comment by SharkFinPro** [comment]
skins arent working :/

**Comment by SharkFinPro** [comment]
NVM, they work now

**Comment by LegitSoulja** [comment]
@SharkFinPro I did mostly all the work and customization on the site. Therefor, if you want it to be special you'll have to do it yourself.

http://refresh-sf.com/

I used this program for most of my scripts to increase speed, and minimize code.

**Comment by SharkFinPro** [comment]
ok

**Comment by rydens** [comment]
Whyyyyy :(

**Comment by rydens** [comment]
Just out of curiosity, do you guys know what the name of the new one will be?

**Comment by LegitSoulja** [comment]
@rydens Yes, but im not allowed to say, nor share.

**Comment by rydens** [comment]
-_- then why even say yes

**Comment by LegitSoulja** [comment]
Because, I'd rather not lie about it.

**Comment by Andrews54757** [comment]
@LegitSoulja By select people, I mean a general audience. But only experienced people. that means no more "I dont know how to do this" complaints

**Comment by Andrews54757** [comment]
The final name of the new one is currently not decided but its code name is

**Comment by Andrews54757** [comment]
For it, I already did most of the work with my large collection of new npm modules I created

**Comment by rydens** [comment]
Auuugggghhhh the suspension
:P

**Comment by Andrews54757** [comment]
CLI-GUI = server control graphical interface
Quadmap = quadmap for server
quickmap = a very fast key-node map
languages = language support
config = config parser

**Comment by rydens** [comment]
sounds awesome

**Comment by Andrews54757** [comment]
I find it strange that @LegitSoulja  and @rydens responded spontaneously to my comments

**Comment by rydens** [comment]
We are simply excited

**Comment by rydens** [comment]
at least, I am

**Comment by LegitSoulja** [comment]
@Andrews54757 lol, my tab is open and it alerts me.

**Comment by rydens** [comment]
Same
while I try to completely finish [KoalasToTheMax](http://www.koalastothemax.com/)
lol

**Comment by LegitSoulja** [comment]
@rydens you didn't even make that

**Comment by rydens** [comment]
I know
I meant get the picture fully formed...

**Comment by LegitSoulja** [comment]
ohh, i see.

**Comment by rydens** [comment]
wait
so whats the "codename"

**Comment by SharkFinPro** [comment]
@rydens New OgarUL
XD

**Comment by LegitSoulja** [comment]
@rydens it will be revealed shortly

**Comment by rydens** [comment]
well derp
lol

**Comment by Andrews54757** [comment]
I woulnt say shortly, prabably after 2016

**Comment by LegitSoulja** [comment]
Well that'll be never due to the issues being closed. 

And plus, I thought it would be sercret?

**Comment by rydens** [comment]
Maybe issues won't be closed, just all of the dev team will ignore them.
A suggestion:
You all simply ignore all of these issues, but keep them active, so members of the still-existing OgarUL community can help each other out,

**Comment by LegitSoulja** [comment]
The issue will get out of hand when the client goes down. 

Just like taking candy from a bunch of kids they'll go insane.

**Comment by rydens** [comment]
true...

**Comment by SharkFinPro** [comment]
you can designate a few people that can moderate

**Comment by rydens** [comment]
Which mainly just means "delete old issues" :P

**Comment by LegitSoulja** [comment]
You can't delete issues. You'll have to delete comment after comment after a comment. It's a waste of time for having someone deal with the issues when they don't even own this project.

**Comment by rydens** [comment]
I mean close the issues

**Comment by Andrews54757** [comment]
I have my ways. so don't fret

**Comment by Andrews54757** [comment]
@LegitSoulja Look at your bitter

**Comment by rydens** [comment]
He has his ways >:)

**Comment by Agarian** [comment]
This god forsaken project has been going downhill ever since that damn charity deadline!

**Comment by LegitSoulja** [comment]
@Andrews54757 checking now

**Comment by AlexandriteErikParacelsus** [comment]
This is exciting

**Comment by Agarian** [comment]
This is not exiting -_-

**Comment by AlexandriteErikParacelsus** [comment]
For me it is

**Comment by systemroot** [comment]
After this gonna go "Bye Bye", 
What will happen with OgarUL. I'm talking for this repo. will you delete or just will not give more support/releases?

**Comment by Andrews54757** [comment]
i wont delete. i wont support

**Comment by Agarian** [comment]
Give me ownership or something

**Comment by David1ali12** [comment]
lmao rip I Had fun with it thx @Andrews54757

**Comment by Andrews54757** [comment]
I like to keep my expiramental projects big and short lived.

**Comment by Agarian** [comment]
Dude at least keep issues if you won't give me ownership of this respiratory!

**Comment by LegitSoulja** [comment]
It's annoyance mainly

**Comment by makansahurbebas** [comment]
Hello,
um... Okay, I'm going to straight to the point...
So I wonder, could I follow the new project? It seems cool hehehehe

**Comment by SharkFinPro** [comment]
Hey @Agarian Could you help me with a native canvas game engine?

**Comment by LegitSoulja** [comment]
@SharkFinPro Agarian doesn't know much about the client. He doesn't work on it.

**Comment by LegitSoulja** [comment]
I'm actually re-writing the entire client (es6) just for the hell of it. It's more organized, and faster.

**Comment by rydens** [comment]
have fun

**Comment by LegitSoulja** [comment]
@rydens im already done

**Comment by rydens** [comment]
yay

**Comment by Andrews54757** [comment]
@LegitSoulja The site won't be limited to people. the server software will be. I am thinking about socket.io though

**Comment by Andrews54757** [comment]
What do you think?

**Comment by SharkFinPro** [comment]
@Andrews54757 Socket.io is super easy to make bots :/ You might not want to make it

**Comment by rydens** [comment]
But I assume the new OgarUL will be bot-protected?

**Comment by Andrews54757** [comment]
Well, thats kindof the point, to make it easy to communicate. but i will make it so only user-authorised programs can connect

**Comment by SharkFinPro** [comment]
oh, ok
I made a chat using socket.io and jquery, and bots were super easy, same with spam :P

**Comment by LegitSoulja** [comment]
@Andrews54757 have you looked into uws? I heard it's good, but I don't know how good. socket.io, eh, i guess ok. I've worked with it, instead of websocket.

**Comment by Agarian** [comment]
We've raised $100 only =/
Thats actually more money than i make in a year

**Comment by LegitSoulja** [comment]
I make roughly 14k a year :/

**Comment by LegitSoulja** [comment]
@Andrews I'll just let you figure out what best. Keep discovering. For me, I'm almost done with the client, and I'm slowly re-creating ogarul using another language. 

Btw, I am using websocket(Microsoft.Websocket), but I might just use TCP servers instead for better security.

**Comment by Agarian** [comment]
@LegitSoulja Well you probably have a job.

**Comment by LegitSoulja** [comment]
@Agarian time consuming. I'm not rushing with the re-creation because if I make it it will have to be perfect the first time because I'd rather not have to re make it better. 

So, this may not even be released, until like next year when im finished with it. I have alot of other projects im working on that I need to catch up on, sooo.

**Comment by Andrews54757** [comment]
@LegitSoulja Im starting to wonder, should I continue creating the new version? I also now like creating npm modules. If I create the new version, people will go crazy asking for it, and I don't want to do support, so I don't know

**Comment by LegitSoulja** [comment]
@Andrews54757 Yes you should, OgarUL has a lot of potential for the next big popular Ogar. I agree we are, but we aren't that big to collaborate with eachother.

I thought of just having the forum instead, and allow those who register to the forum will have download access. The forum would be the main place for ogarul, and community, and support. 

Doing this, we could also come across other well known nodejs developers who would like to help with the project incase you decide to stop supporting, but the forum, and team will be there on top of it. Atleast we have full control over the things we do, or allow, and it won't be public to those who don't obtain without being part of the community. You say keys right? Provide keys for one machine one ogarul use. Just so people can't take, and share the project & key ex YouTube. 

I don't know though.

**Comment by Andrews54757** [comment]
@LegitSoulja I have already created key-gen material locally on my computer (so no one can access now) so that is what i am gonna do anyway

**Comment by Agarian** [comment]
Agar.io is losing it's popularity. Give it up. I know i will.

**Comment by LegitSoulja** [comment]
@Agarian who said agar.io was like ogarul?

**Comment by Agarian** [comment]
Hell, I'd  pick diep.io over agar.io any time.

**Comment by Andrews54757** [comment]
Ogarul is not dying because of multiogar Ogarul is dying because of its popularity. I am too busy for little people saying "uh, the command node is not found, doi"

**Comment by Andrews54757** [comment]
Like I said, ogarUL was experimental. Just to teach me and others js

**Comment by Andrews54757** [comment]
And I don't think handing the project over to others is a good idea, than that other person is burdened, even if they want to, they still end up wasting time

**Comment by Andrews54757** [comment]
Now, I create useful and advanced things that put to use my skills. Rather than just creating a game.

**Comment by Megabyte918** [comment]
Agar.io along with private servers are losing popularity, but I don't think they will be going anywhere anytime soon.

**Comment by rydens** [comment]
Please do not abandon OgarUL.
So many, many people use this, just most simply do not say so.
As you said before, the new version would have no support for questions such as
> uh, the command node is not found, doi
Also, if you do hand the project over to someone else, it won't be wasting their time if the WANT to do it.
agario was a popular game, and so many people still play it.
This project extended its functionality so much, and shouldn't just be abandoned.

**Comment by Agarian** [comment]
Do you think he is gonna stop the bullshit that's about to go down? Nice try, but he's to committed for that,  and yeah, it sucks.

**Comment by LegitSoulja** [comment]
Ogarul isn't entirely being abandoned if this repo shuts down for support. Either if there wasn't, or won't be another ogarul I've been working on a new ogarul project (Not using nodejs) which I'm calling NextGenOgar. This will take place of this project.

Due to letting someone else have the project is a no. Andrew won't make nothing, then abandon it and let whoever have it as there's. He will have to be there either or. Why give up something so special or to make it again than to have it for yourself when you made it? 

The reason why support will be to limited is because nobody actually never had issues, or problems. It was mainly the people coming here asking so many questions, and not understanding how nodejs works which we don't cover. Look at the one thousand + issues. 70% of them are trash. There's really no support needed, but I'd rather have a community for ogarul to share thoughts, and collaborate with eachother just for the hell of it incase Andrew might want to ask the people who uses or plays ogarul questions, and or polls that needs to be rated.

Babysitting, is what I'm going to call it. This project was experimental anyways. I don't even know why the issues are open anyways. Maybe to bring in more help that was needed in the past? 

Besides that, if this project do go abandoned and not recreated you might as well say ogarul never existed. If so this project will be recreated then don't worry, you might could just get your hands on the new edition. But nobody shouldn't worry. Atleast you'll still be able to play on ogarul servers. Maybe not host but you're still close to ogarul instead of it being 12 feet underground and never heard of again.

It takes time, and effort to make something nice. Especially since the work is non profit then you might want to limit who gets what because other people are apathetic, and selfish.

If I sowhat had and made this project I probably would've never made it public. Most of my projects I've made aren't even public, but are limited to those who knows me, and have access to my site for my things. 

That's kindof why I thought of the forums Xen to be specific which ogarul community can go there instead, and only a selected few members that registered will have access to the content. Maybe make a way for people to unlock it and download it. Make people do something for you in order to obtain it. 

There's many many ways ongoing to doing this to make it fair for everyone else. I doubt Andrew wants to remove this project on purpose, but it's for the better. For him and everyone else.

**Comment by Andrews54757** [comment]
@LegitSoulja Well said.

**Comment by Andrews54757** [comment]
As for @Agarian 


@Agarian You have been a member for a long time, But now I realize you are very selfish. All you care about is your own enjoyment. Also, I note how you use profanity. You also don't deserve being in ads-dev in reality. We need advanced programmers, not people who just edit configs and minor things. You can't even fix a simple error.  Still, it saddens me to have to ban you from AJS-dev. You are dismissed.


OFFENSE:
> Profanity, rudeness, selfishness


Today I hereby declare you BANNED from AJS-Development
> Banned on 9/14/16

**Comment by Andrews54757** [comment]
Since people are acting rude to me shutting this down I will clearly state my reasons.


### It is a waste of peoples time

* Most issues that we deal with, are not real ones but are fake such as "What is nodejs' or "couldn't find module"
* Other issues, are not even issues or belong here, they are a waste of time. Such are the ones where all people do is chat. My project is not for CHATTING.
* Also, people who use it are in the majority still in school. I know how you play in school and when the teacher comes, you say you are "programming". It is not and it is just a distraction from your classes.
* Lastly, game creation doesn't really achieve anything in general. And the community has gone sour, bullying each other online. Ogarul was definatly not made for bullying

**Comment by Megabyte918** [comment]
@Andrews54757 quick question.. what is Project-Nehemiah? Is that going to be the new project? Or will it be NextGenOgar? I'm confused

**Comment by LegitSoulja** [comment]
@Megabyte918 From what I've seen, and from what I've heard the new name has not even been thought of yet. And why would it be NextGenOgar?

With NextGenOgar I'm creating I'm branching all the Ogar projects into 1. Meaning from NextGenOgar you'll be able to launch any server and make it private, public whatever. The 2 that I'm working on now is the base of Ogar, then next ogarul with all the features. 

The after that, I can make even more types of Ogar servers. No telling when I'll be finished.

**Comment by LegitSoulja** [comment]
NextGenOgar is pretty much a hosting GUI. Currently for Windows.

**Comment by retslac** [comment]
@Agarian banned!???  Man, that just broke my heart.  Don't think he's being treated fairly after all the dedication he put into this.  Especially since he's been part of the community for so long.

I'm not part of AJS, so i don't wanna interfere with any decision made here.  Only reason i'm writing this is because i think a lot of people think this way and i wanted to express my sympathy for @Agarian.  He has my support.  Really think he deserves it.

**Comment by RadicalCoder56** [comment]
i think he deleted his account because i get 404 when i try to access it https://github.com/Agarian

**Comment by LegitSoulja** [comment]
@locktrap Yes. https://github.com/AJS-development/Ogar-unlimited/issues/1461#issuecomment-247217490

**Comment by NatsuTheGreat** [comment]
Is this ban the reason why Agarian left?


Btw, sorry for the off topic comment, but.. @retslac How ya doing? Haven't seen you in very long. Please respond back to this message and/or message me on discord.

**Comment by locktrap** [comment]
probably is :(

**Comment by SharDX** [comment]
Hey @Andrews54757  can i get invited to your future projects  ? thanks

**Comment by Andrews54757** [comment]
What are your skills?

**Comment by Andrews54757** [comment]
I see you do python

**Comment by LegitSoulja** [comment]
@Andrews54757 You should make a quiz. 20 question apon testing skills.

**Comment by locktrap** [comment]
i think i was wrong to ban agarian over an opinion. i side with retslac here. i'm all ears for those who agree.

**Comment by Andrews54757** [comment]
@LegitSoulja Yep, I should

**Comment by Andrews54757** [comment]
If you want to join AJS-dev, you must do https://docs.google.com/forms/d/e/1FAIpQLScBaBGdBQoH7h1xn3kZbx9NZy68BferKQcc177bD134cYTQYQ/viewform

**Comment by SharkFinPro** [comment]
Can Anyone Join? I would like to...

**Comment by LegitSoulja** [comment]
@SharkFinPro if you do, provide full details of your answer. We will know if you used google or not to get your answers....

Also, there should be a limit 1 to how many times people can take the quiz, (If possible with google forms)

**Comment by rydens** [comment]
There is a setting to make it so only one person can take it once, but I think it requires you to sign into your google account to take it.

**Comment by Andrews54757** [comment]
@SharkFinPro You failed: failed on some questions

**Comment by LegitSoulja** [comment]
@Andrews54757 that quiz is so easy. I don't even know how he even failed. Can you share the spreadsheet? Im just curious to what his answers was.

**Comment by SharkFinPro** [comment]
@Andrews54757 What questions did i fail on?

**Comment by SharkFinPro** [comment]
@Andrews54757 I tried again...

**Comment by Andrews54757** [comment]
You guys both got 

```

function ERROR () {
  var a = fs.readFileSync("error.js","utf8")
  b = JSON.parse(a)
  if (b) {
    b.f = hello;
  }
  var l = (b.f) b : false;
  
  console.log("You have successfully fixed the program! Quote from andrew "Good job!"   ")
}
ERROR();
```
 

Wrong. Remember, you can't remove functionality. like what @NatsuTheGreat did by removing ``fs.readFileSync``. Also, your syntax is wrong, also it doesn't have any errors, it isn't the answer I want as it removes functionality @LegitSoulja Care to enlighten them?

**Comment by Andrews54757** [comment]
This tests you on:


* JS syntax
* Escaping
* Error handling 
* and functions

**Comment by LegitSoulja** [comment]
Sure

**Comment by SharkFinPro** [comment]
@LegitSoulja Tell me =)

**Comment by LegitSoulja** [comment]
@SharkFInPro Tell you?

**Comment by SharkFinPro** [comment]
Yes

**Comment by makansahurbebas** [comment]
Maybe this is dumb question but, how do I know when I failed the quiz?

**Comment by Megabyte918** [comment]
OH CRAP @Andrews54757 I answered that question earlier but I didn't realize that "thing" was missing (you know what im talking about) I was skimming and I must have missed it cause It was so small. If you see my answer for that just keep in mind that I didn't realize that "it" was missing at the time

**Comment by Megabyte918** [comment]
@Andrews54757 I think the answer I submitted for that question u mentioned earlier was wrong... can I send you like a gist or something with the corrected code for it?

**Comment by AlexandriteErikParacelsus** [comment]
Agarian deleted his github account.. That's sad

**Comment by LegitSoulja** [comment]
@Andrews54757 Should the client run behind nodejs? It wouldn't be no problem to have nodejs run the client instead of it being a website with pages and scripts.

Would it be better and more secure running behind a vm? My services/websites run behinds a vm and I just thought it would be better to do it that way instead.

I really don't know how you will layout the packers for the client, and I need to know. I'm really kindof stuck right now on what to do, and how to fix certain things with the client because of my lack of physics, and client API knowledge.

**Comment by locktrap** [comment]
Agarian is still alive, ya wanna know why? Because you're lookin' at him.

**Comment by LegitSoulja** [comment]
@locktrap Didn't you want to leave anyways? Why complain?

**Comment by locktrap** [comment]
I wanted to leave because this is the only project i enjoyed, and i lost it.

**Comment by LegitSoulja** [comment]
So you would leave your parents because you enjoyed them so much?

You don't make any sense.

**Comment by locktrap** [comment]
Read the whole comment -_-

**Comment by SharkFinPro** [comment]
:( agarian :(

**Comment by locktrap** [comment]
Yes, i know.

**Comment by locktrap** [comment]
You didn't think i'd just die did you sharkfin?

**Comment by LegitSoulja** [comment]
@locktrap ok, since you edited your comment it makes sense now.

but still, you wanted to leave, and your wish came true.

**Comment by locktrap** [comment]
Like it mattered, the project is still going down

**Comment by LegitSoulja** [comment]
@locktrap 
> Like it mattered

Exactly, nothing matters to you. And this project isn't going down. Only the impatient ones will suffer. Even if this project does go down then what? Nothing really won't happen. The project will be limited to certain people anyways.

**Comment by locktrap** [comment]
The bottom line is, I got banned but didn't actually die like I said.

**Comment by LegitSoulja** [comment]
Wanting someone to die is the last thought in some peoples head. Unless you actually have suicidal thoughts, then you really need some help.

**Comment by LegitSoulja** [comment]
There's a suicide hotline. Maybe you should call them and talk about your issues dealing with life, since this project shouldn't be one of them.

**Comment by locktrap** [comment]
You know I did not mean die like that -_-

**Comment by locktrap** [comment]
Bye.

**Comment by LegitSoulja** [comment]
Call them, that's all I have to so. Good day.
